#16898 - extensions to test scripting language (#16899)
prometheus/prometheus: promql/promqltest/README.md

* `<start>` and `<end>` specify the time range of the range query, and use the same syntax as `<time>`
* `<step>` is the step of the range query, and uses the same syntax as `<time>` (eg. `30s`) 
* `<expect>`(optional) specifies expected annotations, errors, or result ordering.
* `<expect range vector>` (optional) for an instant query you can specify expected range vector timestamps
* `<series>` and `<points>` specify the expected values, and follow the same syntax as for `load` above

> Note - a string literal instant series is now suppported. The string literal can be defined by placing the string within quotes (""). 

Reviewer: [nit] I would document what is supported, rather than what's changed:

```suggestion
String literals results are supported. The string literal can be defined by placing the string within quotes (""). 
```

This is only valid for an instant query.

For instance;

```
eval instant at 50m ("Foo")

> 	"Foo"

Reviewer: If we're going to use `expect range vector` for range vectors, what if we had syntax like this for strings? Then it would be consistent(ish) with the range vector and error expectation syntax, and we can ignore any issues with quoting, escaping etc.:

```
eval instant at 50m ("Foo")
	expect string: Foo
```
